apis/discoveryengine/v1alpha1/discoveryenginedatastore_types.go

```diff
@@ -67,6 +67,10 @@ type DiscoveryEngineDataStoreSpec struct {
 	/* The ID of the project in which the resource belongs.*/
 	ProjectRef *refs.ProjectRef `json:"projectRef"`
 
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Collection field is immutable"
+	// Immutable. The collection for the DataStore.
+	Collection string `json:"collection"`
```

Comment:
I realized that parent is not suitable for immutable CEL because this can be tricked, and we already validate the parent immutability from the `status.externalRef`

Reply:
Not sure what you mean.  Are you saying we should not have the immutability CEL rule here?

Reply:
Yes.